lightning/src/blinded_path/message.rs

```diff
@@ -138,17 +136,25 @@ pub(super) fn blinded_hops<T: secp256k1::Signing + secp256k1::Verification>(
 ) -> Result<Vec<BlindedHop>, secp256k1::Error> {
 	let pks = intermediate_nodes.iter().map(|node| &node.node_id)
 		.chain(core::iter::once(&recipient_node_id));
-	let tlvs = pks.clone()
+	let tlvs: Vec<ControlTlvs> = pks.clone()
 		.skip(1) // The first node's TLVs contains the next node's pubkey
 		.zip(intermediate_nodes.iter().map(|node| node.short_channel_id))
 		.map(|(pubkey, scid)| match scid {
 			Some(scid) => NextMessageHop::ShortChannelId(scid),
 			None => NextMessageHop::NodeId(*pubkey),
 		})
 		.map(|next_hop| ControlTlvs::Forward(ForwardTlvs { next_hop, next_blinding_override: None }))
-		.chain(core::iter::once(ControlTlvs::Receive(ReceiveTlvs{ context: Some(context) })));
+		.chain(core::iter::once(ControlTlvs::Receive(ReceiveTlvs{ context: Some(context) })))
+		.collect();
 
-	utils::construct_blinded_hops(secp_ctx, pks, tlvs, session_priv)
+	let max_length = tlvs.iter()
+		.max_by_key(|c| c.serialized_length())
+		.map(|c| c.serialized_length())
```

Comment:
I think we could do:

```rust
.map(|tlv| tlv.serialized_length())
.max()
```

Reply:
Followed the suggestion from one of your following comments!
Thanks a lot for the pointers!

Reply:
Should be able to do this, too?

Reply:
Yep! This works! The code's now a lot cleaner! :)
Addressed in [pr3177.04](https://github.com/shaavan/rust-lightning/commits/pr3177.04)
Thanks a lot for the suggestion!